### ISSUE TITLE
[Agent] Unable to Stop elastic-agent service from Services manager.

### ISSUE THREAD

**Issue by vipulsrivastava-qasource** [elastic/beats#27527]
**Description:**
Unable to Stop elastic-agent service from Services manager.

**Build Details:**

Version: 7.15.0-BC1
Build number: 43636
Commit: d791226d9385122f33f4a5ca38fa5369012fbec3
Artifact: https://staging.elastic.co/7.15.0-d9929120/summary-7.15.0.html

**Browser Details:**
All

**Preconditions:**

1. 7.15.0 BC1 Kibana user should be logged in.
2. Window Agent deployed with system Integration. 

**Steps to Reproduce:**

1. RDP to the window agent.
2. Stop the elastic-agent service from services manager.
3. Observe that elastic-agent service does not stop.

**Impacted Test case:**
N/A

**Actual Result:**
Unable to Stop elastic-agent service from Services manager.

**Expected Result:**
Elastic agent service should stop after click on stop the elastic-agent service from Services manager.

**Screen Recording**

https://user-images.githubusercontent.com/69579402/130224246-24aa7623-aee0-4832-8a2f-f04932937fc4.mp4

**Comment by vipulsrivastava-qasource** [comment]
@manishgupta-qasource : Please review the Bug.

**Comment by manishgupta-qasource** [comment]
Reviewed & assigned to @andresrc

**Comment by EricDavisX** [comment]
Hi, @blakerouse not sure if you have time yet to poke this, does it feel related to the recent delay-enrollment feature work?

**Comment by blakerouse** [comment]
@EricDavisX I do not feel its because of the delay enrollment feature.

Can the logs of the Elastic Agent on that host be attached to this issue?

It also says the program exited unexpectedly, but then still shows that it's running. I wonder if you refresh the services.msc if it would then show as stopped.

**Comment by vipulsrivastava-qasource** [comment]
Hi @blakerouse 

Please find the logs for Elastic-Agent and let me know if anything else is required from our end.
[logs.zip](https://github.com/elastic/beats/files/7036600/logs.zip)

Thanks!!

**Comment by EricDavisX** [comment]
@dikshachauhan-qasource @amolnater-qasource adding to high impact list.

**Comment by blakerouse** [comment]
Dang those logs are not helpful, because it shows in the logs that the Elastic Agent stopped correctly. It does not log an error or anything and shows that it shutdown cleanly.... strange.

**Comment by EricDavisX** [comment]
@vipulsrivastava-qasource @amolnater-qasource @dikshachauhan-qasource can you confirm if you refresh the view to help confirm if it is actually in the correct state?

**Comment by blakerouse** [comment]
I am seeing the same behavior on Window 10. But I do see that the Elastic Agent does stop, but it does show that error message about stopping `terminated unexpectedly`.

Can you check that the Elastic Agent does in fact stop? Click the refresh button in the `services.msc` to ensure that the view is updated with the Elastic Agent stopped.

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have attempted to stop agent from services on Windows Server 2012 R2[VSphere], Windows 2019 base[AWS], Windows 8.1[VSphere] and Windows 10 x64[VSphere].
- We are **unable to stop** elastic-agent from services on above mentioned OS's.

**Build details:**
7.15.0 BC-3 Kibana Cloud environment
Build: 43818
Commit: 6f7562b1906dcfad65809da8fdec15df353d0252 
Artifact Link: https://staging.elastic.co/7.15.0-642a73fa/downloads/beats/elastic-agent/elastic-agent-7.15.0-windows-x86_64.zip

Please let us know if anything else is required from our end.
Thanks

**Comment by michalpristas** [comment]
my idea of  this behaviour is that agent shuts down before signaling service manager. SM then starts the agent as it understands this as a crash and we (service manager) restart on crash

**Comment by dikshachauhan-qasource** [comment]
Hi @EricDavisX 

We confirm you that above reported behavior is reproducible on AWS windows server 2016 base machine too. 

On attempting to stop the agent, below message shows up, however state continues to show up as running.
![image](https://user-images.githubusercontent.com/72375798/131836956-690753ac-7d95-46a4-8f8c-3df80d193501.png)


Further, we also observed agent going to unhealthy state after triggering action with below errors:
```
16:53:43.865
elastic_agent
[elastic_agent][warn] Elastic Agent status changed to: 'degraded'
16:53:43.865
elastic_agent
[elastic_agent][info] 2021-09-02T11:23:43Z - message: Application: metricbeat--7.15.0[4a17211e-df09-4c52-a70b-6c21bf6bc446]: State changed to DEGRADED: Missed last check-in - type: 'STATE' - sub_type: 'RUNNING'
16:54:43.917
elastic_agent
[elastic_agent][error] Elastic Agent status changed to: 'error'
16:54:43.922
elastic_agent
[elastic_agent][error] 2021-09-02T11:24:43Z - message: Application: metricbeat--7.15.0[4a17211e-df09-4c52-a70b-6c21bf6bc446]: State changed to FAILED: Missed two check-ins - type: 'ERROR' - sub_type: 'FAILED'
```

Also, after some time it resumes it state back to healthy.

Screenshot:
![image](https://user-images.githubusercontent.com/72375798/131837106-25c5e3e7-bf26-4870-a068-926982701694.png)

Agent logs: 
[logs.zip](https://github.com/elastic/beats/files/7098554/logs.zip)

Please let us know if anything further is required.

Thanks
QAS

**Comment by michalpristas** [comment]
i did a small experiment and my initial idea seems true. 
i will try to find a good/better way to fix this.

**Comment by EricDavisX** [comment]
backport is done, it will be available in the next 7.15 BC
